mlflow/xgboost.py

```diff
@@ -286,6 +287,30 @@ def autolog(importance_types=["weight"]):  # pylint: disable=W0102
     import xgboost
     import numpy as np
 
+    class _InputExampleInfo:
+        def __init__(self, input_example=None, error_msg=None):
+            self.input_example = input_example
+            self.error_msg = error_msg
+
+    def __init__(self, *args, **kwargs):
+        data = args[0] if len(args) > 0 else kwargs.get("data")
+
+        if data is not None:
+            original = gorilla.get_original_attribute(xgboost.DMatrix, "__init__")
+
+            input_example_info = None
+            try:
+                if isinstance(data, str):
+                    raise Exception("The input data was of type string.")
```

Comment:
Can we be more specific about the types of input data that are supported / indicate why we can't capture string data? More specifically, the problem is that `data` refers to a path, right?